### PR TITLE
Refactor `sync.py` module

### LINKED ISSUE #52: Refactor `sync.py` module
Especially after merging #40 there will be a lot of redundant and convoluted functionality that we should delete.

Examples:
* `download_dataset` and all the functions it is calling  (e.g. `get_transform`). Do not forget to remove all the references in `docs`.

### PR DESCRIPTION
**Pull request by jankrepl**
Closes #52 

# What was done
* Remove everything that is not directly used inside of `download_dataset_parallel`
* Remove corresponding tests

See the coverage after this PR
```bash
---------- coverage: platform darwin, python 3.8.11-final-0 ----------
Name                     Stmts   Miss Branch BrPart  Cover   Missing
--------------------------------------------------------------------
src/atldld/__init__.py       2      0      0      0   100%
src/atldld/base.py         101      0     32      0   100%
src/atldld/cli.py           20     20      2      0     0%   17-61
src/atldld/sync.py          35      0     12      0   100%
src/atldld/utils.py        121     35     48      5    67%   57, 114, 126, 131-151, 170-184, 512, 532-552
src/atldld/version.py        1      0      0      0   100%
--------------------------------------------------------------------
TOTAL                      280     55     94      5    79%
```

### CODE REVIEW

**Comment by jankrepl** [comment]
@Stannislav @EmilieDel Feel free to review

File: docs/source/allen.rst [hunk 1]
```diff
@@ -70,20 +70,21 @@ Single image download
 
 Full-blown registration
 ~~~~~~~~~~~~~~~~~~~~~~~
-The most useful function is the :code:`download_dataset`. It expects the user to specify the unique
-:code:`dataset_id`. It returns a generator. Each iteration then yields
+The most useful function is the :code:`download_dataset_parallel`.
+It expects the user to specify the unique :code:`dataset_id`. It returns a
```
Comment (EmilieDel) [inline docs/source/allen.rst:74]: Just a suggestion, what do you think about explaining why we call this `parallel` ? I just checked and it is in the docstring but maybe we can be more explicit about it. However, I don't mind keeping it that way!

**Review by EmilieDel — APPROVED** [review]
That is what we call cleaning :) LGTM! Thanks @jankrepl

File: docs/source/allen.rst (hunk 1 above)
Comment (jankrepl) [reply docs/source/allen.rst:74]: Good point:) I will add a short description. Note that I am also planning to dedicate a new PR to documentation of this functionality #33

Comment (jankrepl) [reply docs/source/allen.rst:74]: Added some text here e6af21ba872778c52788356df1415d1e27bf254f

File: README.md [hunk 2]
```diff
@@ -82,9 +82,11 @@ dataset_ids = get_experiment_list_from_gene("Pvalb", axis='sagittal')
 - One can download any dataset from a dataset ID:
 ```python
 from atldld.sync import download_dataset_parallel
-dataset = download_dataset(dataset_id=DATASET_ID, **kwargs)
+dataset = download_dataset_parallel(dataset_id=DATASET_ID, **kwargs)
```
Comment (Stannislav) [inline README.md:85]: Suggestion: rename `download_dataset_parallel` to `download_parallel_dataset`.

Why? The former sounds like it's a multi-threaded dataset download or something (unless you're Yoda 😜) Same for `get_transform_parallel`.

What do you think?

File: docs/source/allen.rst [hunk 3]
```diff
@@ -70,20 +70,21 @@ Single image download
 
 Full-blown registration
 ~~~~~~~~~~~~~~~~~~~~~~~
-The most useful function is the :code:`download_dataset`. It expects the user to specify the unique
-:code:`dataset_id`. It returns a generator. Each iteration then yields
+The most useful function is the :code:`download_dataset_parallel`.
```
Comment (Stannislav) [inline docs/source/allen.rst:73]: Can we discuss the difference between
- `` :code:`stuff` ``
- `` `stuff` ``
- ``` ``stuff`` ```

?

According to the [some docs](https://docutils.sourceforge.io/docs/user/rst/quickref.html#inline-markup)
- `` `stuff` `` = interpreted text
- ``` ``stuff`` ``` = inline literal

Usually when I want monospaced text in `rst` I go for ``` ``stuff`` ```. Is `` :code:`stuff` `` going to add some additional functionality/highlighting?

Is `` :code:`stuff` `` part of the `rst` syntax or is it specifi to Sphinx?

**Review by Stannislav — COMMENTED** [review]
More deletions! Feels good 😆 

Anyway, left two small comments.

File: README.md (hunk 2 above)
Comment (jankrepl) [reply README.md:85]: hehe:) I agree! I will rename it:)

File: docs/source/allen.rst (hunk 3 above)
Comment (jankrepl) [reply docs/source/allen.rst:73]: Hmm. I actually always go for `` :code:`whatever` `` when I want to enforce code highlighting. I did not know that ` ``stuff``  ` has exactly the same effect! 

Assuming they are identical, ` ``stuff`` ` is probably the less verbose option and I am totally fine with using it everywhere. What do you think?

File: README.md (hunk 2 above)
Comment (jankrepl) [reply README.md]: Done in fd2a84061a80dc976ba03b17631ccff0ef88e0f8

File: docs/source/allen.rst (hunk 3 above)
Comment (jankrepl) [reply docs/source/allen.rst]: Anyway, this PR did not introduce this syntax. I propose we create an issue and that we do not try to solve it in this PR.

Comment (Stannislav) [reply docs/source/allen.rst]: Good idea. Issue: #61

**Review by Stannislav — APPROVED** [review]

**Pull request merged**